play.pokemonshowdown.com/js/client-chat.js

```diff
@@ -357,43 +357,73 @@
 				var m1 = /^([\s\S]*?)([A-Za-z0-9][^, \n]*)$/.exec(prefix);
 				var m2 = /^([\s\S]*?)([A-Za-z0-9][^, \n]* [^, ]*)$/.exec(prefix);
 				if (!m1 && !m2) return true;
+				var cmds = this.tabComplete.commands;
+				var shouldSearchCommands = !cmds || !cmds.filter(function (x) {
+					return x.startsWith(prefix);
+				}).length;
+				var isCommandSearch = text.startsWith('/') || text.startsWith('!');
+				if (isCommandSearch && shouldSearchCommands) {
+					if (this.tabComplete.searchPending) return true; // wait
+					this.tabComplete.isCommand = true;
+					this.tabComplete.searchPending = true;
+					app.send('/crq cmdsearch ' + text);
+					var self = this;
+					app.once('response:cmdsearch', function (data) {
```

Comment:
Isn't the handler supposed to be defined before the `/crq` is sent?

Reply:
It works either way. But I suppose there's no real harm in being cautious.